cmd/operator-sdk/bundle/build.go

```diff
@@ -0,0 +1,81 @@
+package bundle
+
+import (
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	dirBuildArgs       string
+	tagBuildArgs       string
+	imageBuilderArgs   string
+	packageNameArgs    string
+	channelsArgs       string
+	channelDefaultArgs string
+	overwriteArgs      bool
+)
+
+// newBundleBuildCmd returns a command that will build operator bundle image.
+func newBundleBuildCmd() *cobra.Command {
+	bundleBuildCmd := &cobra.Command{
+		Use:   "build",
```

Comment:
I think it should be something as `operator-sdk olm-catalog gen-csv --csv-version 0.1.0 --create-bundle`

Following why I think it.  
- The --csv-version 0.1.0 is required to know the version which should be used to create the bundle
- I understand that the CSV is part of the bundle  according to the info in https://operatorhub.io/bundle  and https://github.com/operator-framework/community-operators

Reply:
Hey Camila,

Thank you for the initial review. I would like to add the bundle enhancement and bundle library docs so you can take a look and have a better understand on what my team is trying to do with this PR. I should've added those docs on commit message but I forgot. My apology.

- [Bundle Enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md)

- [Bundle Library Doc](https://github.com/operator-framework/operator-registry/pull/97/files?short_path=ba4fcb6#diff-ba4fcb6a5514edd925d1960cee5e7e33)

One note that the PR for adding bundle package to operator-registry is still pending-merge so that's why this PR fails some tests due to the package is not available yet.

The goal of this PR to add a new `bundle` command under the top-level `operator-sdk` command. The `bundle` command itself has two subcommands to `generate` bundle metadata and `build` bundle container image.  It should strictly use bundle package/library to perform its tasks without interfering with any existing functionalities of SDK. This PR shouldn't change or impact any existing SDK commands/subcommands. In a way, this is a "plug-in" that is independent in some way. I do not wish to nest/add/wrap this bundle command tool under other existing SDK commands as it is a brand-new feature and if we do so, it will affect other existing commands which I believe something Joe doesn't want. Secondly, bundle command requires a few flags that if nested under other commands will create very messy UX as the commands will be too longer and confusing especially if the command contains something that has nothing to do with creating bundle image such as `olm-catalog` or even `olm` itself. OLM is supposed to be a consumer of bundle images, not creator.

If you have any further questions, please feel free to ask. :)

Reply:
Hi @dinhxuanvu,

Thank you for share the info. Following my main concerns/review on it. 

IMHO

- it should follow up the same idea/standard used in the other SDK commands 
- it should by default use the SDK structure project and the flags are optional instead of mandatory 
- It is not passing in the tests
- it is missing docs, tests, changelog
- it shows missing the impl of each command.